app/flux/components/Checkout.react.js

```diff
@@ -0,0 +1,96 @@
+var React = require('react');
+
+var { State } = require('react-router');
+var StoreAPI = require('../API/StoreAPI');
+
+var UserStore = require('../stores/UserStore');
+
+class Checkout extends React.Component{
+    constructor (props){
+        super(props);
+
+        // Only lifecycle methods receive the inheritance from React.Component.
+        this._onChange = this._onChange.bind(this);
+        
+        this.state = {
+            stripe: null,
+            training: null,
+            user: UserStore.user        
+        };
+    }
+
+    componentDidMount (){
+        UserStore.addChangeListener(this._onChange);
+
+        if (!this.state.user && UserStore.isPending){
+            setTimeout(() => {
+                if (!this.state.user){
+                    this.context.router.transitionTo('signup');    
+                }
+            }, 800).bind(this);
+        } else if (!this.state.user){
+            this.context.router.transitionTo('signup');
+        }
+
+        var trainingId = this.context.router.getCurrentParams().id;
```

Comment:
Just a personal preference, but I prefer to pass down the params as ```...props``` and then just use ```this.props.params.id```.